raft/rafttest/interaction_env_handler.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafttest
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/cockroachdb/datadriven"
+)
+
+func (env *InteractionEnv) Handle(t *testing.T, d datadriven.TestData) string {
+	env.Logger.Buf.Reset()
+	buf := env.Logger.Buf
+	var err error
+	switch d.Cmd {
+	case "add-nodes":
+		err = env.handleAddNodes(t, d)
+	case "campaign":
+		err = env.handleCampaign(t, d)
+	case "compact":
+		err = env.handleCompact(t, d)
+	case "deliver-msgs":
+		err = env.handleDeliverMsgs(t, d)
+	case "process-ready":
+		err = env.handleProcessReady(t, d)
+	case "log-level":
+		err = env.handleLogLevel(t, d)
+	case "raft-log":
+		err = env.handleRaftLog(t, d)
+	case "stabilize":
+		err = env.handleStabilize(t, d)
+	case "status":
+		err = env.handleStatus(t, d)
+	case "tick-heartbeat":
+		err = env.handleTickHeartbeat(t, d)
+	case "propose-conf-change":
+		err = env.handleProposeConfChange(t, d)
+	default:
+		err = fmt.Errorf("unknown command")
+	}
+	if err != nil {
+		buf.WriteString(err.Error())
+	}
+	// NB: the highest log level suppresses all output, including that of the
+	// handlers. This comes in useful during setup which can be chatty.
+	if buf.Len() == 0 || env.Logger.Lvl == len(lvlNames)-1 {
+		return "ok\n"
+	}
+	s := buf.String()
+	if len(s) > 0 && s[len(s)-1] != '\n' {
+		s += "\n"
+	}
+	return s
+}
+
+func mustFirstAsInt(t *testing.T, d datadriven.TestData) int {
+	n, err := strconv.Atoi(d.CmdArgs[0].Key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return n
+}
+
+func ints(t *testing.T, d datadriven.TestData) []int {
```

Comment:
Either rename this to `mustInts` or remove the `must` from `mustFirstAsInt` (for test code, I'd prefer to leave out the `musts`)

Reply:
I removed `must`.